src/test/java/design/subsite/CellTest.java

```diff
@@ -0,0 +1,320 @@
+                 assertNull(cell.getBel(), cell.getName() + " Cell still attached to " + bel.getName() + " after being removed from design.");
+             }
+         }
+     }
+
+     @Test
+     @DisplayName("test Cell method 'getNetList'")
+     public void testGetNetList() {
+         // test each of the sample test cells
+         for (Cell cell : testcells) {
+             // keep track of which nets have been added
+             List<CellNet> addednets = new ArrayList<CellNet>();
+             // keep track of all the cell pins
+             List<CellPin> cellpins = new ArrayList<CellPin>(cell.getPins());
+             // connect each cell pin to a net one at a time
+             for (int index = 0; index < cellpins.size(); index++) {
+                 // create a new net for the pin
+                 CellNet newnet = new CellNet("input_net_" + Integer.toString(index), NetType.WIRE);
+                 // keep track of the new net
+                 addednets.add(newnet);
+                 // connect the new net to the pin
+                 newnet.connectToPin(cellpins.get(index));
+                 // verify the netlist
+                 verifyNetList(cell, addednets);
+             }
+             // remove the nets one at a time
+             for (int index = 0; index < cellpins.size(); index++) {
+                 // disconnect the pin from the net
+                 addednets.get(0).disconnectFromPin(cellpins.get(index));
+                 // remove the net from the expected netlist
+                 addednets.remove(0);
+                 // verify the netlist
+                 verifyNetList(cell, addednets);
+             }
+         }
+     }
+
+     /**
+      * helper function to verify a Cell's netlist
+      * cell the Cell to test
+      * expected the expected netlist
+      */
+     private void verifyNetList(Cell cell, Collection<CellNet> expected) {
+         Collection<CellNet> actual = cell.getNetList();
+         assertEquals(expected.size(), actual.size(), "netlist size mismatch after adding new net to " + cell.getName());
+         for (CellNet net : expected) {
+             assertTrue(actual.contains(net), cell.getName() + " Cell doesn't contain expected net while connecting pins.");
+         }
+     }
+
+     @Test
+     @DisplayName("test Cell psuedo pins")
+     public void testPsuedoPins() {
+         for (Cell cell : testcells) {
+             // create and attach a pseudo pin to the cell
+             CellPin pseudopin = cell.attachPseudoPin("test_pin", PinDirection.INOUT);
+             // make sure the cell has the pseudo pin
+             assertEquals(1, cell.getPseudoPinCount(), cell.getName() + " Cell doesn't contain a psuedo pin after attaching one.");
+             // verify its the right pseudo pin
+             assertEquals(pseudopin, cell.getPseudoPins().iterator().next(), cell.getName() + " Cell doesn't contain the right pseudo pin.");
```

Comment:
You should also test the function `cell.getPin("test_pin")` to make sure the psuedo pin is returned in this case as well (it should be).